libexec/bats-core/bats-exec-suite

```diff
@@ -78,7 +83,7 @@ for filename in "$@"; do
   done < <(BATS_TEST_FILTER="$filter" bats-preprocess "$filename")
 
   if [[ "${#test_dupes[@]}" -ne 0 ]]; then
-    printf 'bats warning: duplicate test name(s) in %s: %s\n' "$filename" "${test_dupes[*]}" >&2
+    abort "Duplicate test name(s) in file \"${filename}\": ${test_dupes[*]}"
```

Comment:
This will immediately stop execution, right?
Did you consider only failing in the end including the failure message as I suggested in the issue?

Reply:
I agree with @vampire. The full test run is still valuable feedback. Marking just the duplicated test as failed is a useful alternative. Though we probably want to distinguish that failure somehow that it's different from a normal test failure.

Reply:
I did — my reasoning was that it's less confusing to error immediately and fail fast, otherwise if all tests pass there's a passing suite but a failing exit code and possibly a missed error.

Reply:
That's why I said the error should also be printed last.
I agree that printing the error above all tests and then having a negative exit code is too quiet, especially as during manual invocation you will easily miss that there was a non-0 exit code.

Either @jasonkarns suggestion to actually fail the tests in question would be a good idea, or printing the errors in the end accompanied with the non-0 exit code, like I did here: https://github.com/svn-all-fast-export/svn2git/commit/ece0088e8f88561c45838ab3b4dfb3d21baa9fa3

Reply:
It feels non-obvious to fail after passing test suite, for a check done in pre-processing. Users should rename the test anyway.

I wonder if there's also a risk of nondeterminism here with parallel `--job`s, the "second" duplicate suite may sometimes run as the first. This wouldn't break anything (we've already got a breaking change for the build failing) but if they give different results is a misdirecting user experience. Needs a test.

It seems faster, simpler, and more obvious to fail when the duplicate is detected. That said we can discuss on a PR if preferred :pray:

Reply:
No, it's fine, if it is a reasoned decision rather than a read over of my comment, it's fine.
Especially as it fails at the very beginning.

Reply:
I'm not too bothered by the early failure, because it denotes a mistake in the test setup itself (as sublimino said), but if we really wanted to be more graceful I think a better fix would be to not key off (only) the user-provided test description, and instead ensure that the generated function name _is_ unique.

Reply:
Which would still make it hard to connect failure to test then, I like it failing for duplicate test names. :-)
While then there could be a configuration option on whether to fail or not if there are duplicate names, while either way at least the tests would run correctly.